scrapy/extensions/feedexport.py

```diff
@@ -485,4 +485,4 @@ def _get_uri_params(self, spider, uri_params, slot=None):
         params['batch_id'] = slot.batch_id + 1 if slot is not None else 1
         uripar_function = load_object(uri_params) if uri_params else lambda x, y: None
         uripar_function(params, spider)
-        return params
+        return uripar_function(params, spider)
```

Comment:
Maybe it is not as straightforward as I thought. Also, I think we may need to be careful to ensure backward compatibility.

I wonder what would happen if we only return the result of `uripar_function` if it is not `None`:

```python
        new_params = uripar_function(params, spider)
        return new_params if new_params is not None else params
```

So, if someone wrote a `uripar` function that returns nothing and instead simply edits the input params, this code should work as well.

Assuming this code works, then we probably just need to extend it to log a deprecation warning if `uripar_function` returns `None`, and we should update our tests to cover both use cases.

Reply:
Thanks for the reply! I will try this later and see how it goes.

Reply:
It seems like `return new_params if new_params is not None else params` passed all the checks. Thanks for the advice! Let me know if there is anything I can help. :)